main.tf

```
@@ -1,13 +1,204 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled                 = module.this.enabled
+  create_log_bucket       = var.bucket_id == null ? true : false
+  bucket_id               = var.bucket_id != "" ? var.bucket_id : module.ssm_patch_log_s3_bucket.bucket_id
+  default_allowed_actions = ["s3:GetObject", "s3:PutObject", "s3:PutObjectAcl", "s3:GetEncryptionConfiguration"]
+}
+
+module "ssm_patch_log_s3_bucket" {
+  count                  = local.create_log_bucket ? 1 : 0
+  source                 = "cloudposse/s3-bucket/aws"
+  version                = "0.38.0"
+  acl                    = "private"
+  versioning_enabled     = false
+  allowed_bucket_actions = var.ssm_bucket_policy != null ? [] : local.default_allowed_actions
+  policy                 = var.ssm_bucket_policy
+  context                = module.this.context
+}
+
+resource "aws_ssm_maintenance_window" "scan_window" {
+  count    = module.this.enabled ? 1 : 0
+  name     = "${module.this.id}-scan-window"
```

```suggestion
  name     = "module.this.id"
```

The resource type is `scan window`, no need to include the resource type in the resource name.
(And if we want to add additional suffix, we use a `label` module with additional attributes)